packages/medusa/src/api/routes/admin/price-lists/update-price-list.ts

```diff
@@ -83,26 +85,54 @@ import { validator } from "../../../../utils/validator"
  */
 export default async (req, res) => {
   const { id } = req.params
+  let priceList
+  const featureFlagRouter = req.scope.resolve("featureFlagRouter")
+  const manager: EntityManager = req.scope.resolve("manager")
+  const priceListService: PriceListService =
+    req.scope.resolve("priceListService")
 
   const validated = await validator(
     AdminPostPriceListsPriceListPriceListReq,
     req.body
   )
 
-  const priceListService: PriceListService =
-    req.scope.resolve("priceListService")
+  if (featureFlagRouter.isFeatureEnabled(MedusaV2Flag.key)) {
```

Comment:
question: should we check for workflow flag here

Reply:
Workflow flags are only for app logic that doesn't involve pricing and product modules, since pricing doesn't have a legacy path working. 

I've removed the workflow flags from other places in pricing path.